### ISSUE TITLE
Larger strides/downsampling factors

### ISSUE THREAD

**Issue by chanshing** [adobe/antialiased-cnns#37]
First, thanks for the very nice work!

In your implementation as well as in the paper, it seems that the proposed filters (which are the [binomial coefficients](https://en.wikipedia.org/wiki/Binomial_theorem)) are only valid for strides/downsampling factors of 2. Extrapolating from this, does it mean that I need to use the [trinomial coefficients ](https://mathworld.wolfram.com/TrinomialCoefficient.html#:~:text=A%20trinomial%20coefficient%20is%20a,given%20by%20the%20closed%20form) for stride 3, quadrinomial coefficients for stride 4, and so on?

By the way, you could simplify your code using [scipy.special.binom](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.binom.html) instead of hard-coding each filter.

**Comment by richzhang** [comment]
Yes the width of the filter can be computed automatically based on downsampling factor. The binom filter shouldn't be used for larger filter sizes and stride 2. I'm working on larger strides

**Comment by chanshing** [comment]
I implemented something for the 1D case, since I'm doing timeseries stuff mostly. The idea is to define the filter order instead of the filter size, and start from the box filter (average pooling) of width equal to the down factor. This will be order 0. Then for the higher order filters we convolve the box filter with itself _order_ times (See http://nghiaho.com/?p=1159)

For example, for stride 2 and order 3, we start with the box filter `[1, 1]`. Convolve with itself once and we get `[1, 2, 1]`. Once more and we get `[1, 3, 3, 1]`. Once again and we get `[1, 4, 6, 4, 1]`. Example with stride 3: `[1, 1, 1] -> [1, 2, 3, 2, 1] -> [1, 3, 6, 7, 6, 3, 1] -> ...`

Here's more or less what I did (again, this is only 1D):
```python
box_kernel = np.ones(factor)
kernel = np.ones(factor)
for _ in range(order):
    kernel = np.convolve(kernel, box_kernel)
kernel /= np.sum(kernel)
kernel = torch.Tensor(kernel)
self.register_buffer('kernel', kernel[None, None, :].repeat((channels, 1, 1)))
```

For 2D and beyond we could take the outer product of the 1D filter with itself.
